data-products/src/freestar_revenue_reporting/freestar_reporting_flow.py

```diff
@@ -162,9 +156,15 @@ def extract_freestar_data(
             if col not in df.columns:
                 df[col] = None  # Add the column with None values
 
+        record_count = len(data)
+
         df.to_parquet(file_path.format(page_number))
```

Comment:
:heavy_check_mark:  This function runs one more time by moving the condition down.